### PR TITLE
Update gettext to latest tagged release

### PR DESCRIPTION
**Pull request by borisson**
screen shot 2018-08-31 at 15 12 42

This improves performance _for free_. By quite a lot if I understand blackfire's results correctly.

This should be tested very well before merging.

### CODE REVIEW

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by mlutfy** [comment]
add to whitelist

**Comment by borisson** [comment]
Yeah, I also found that page earlier, I think that this also provides value though. 

I'm not sure if any of the other libraries need to be updated as well, for better performance/features. But this one stuck out in the blackfire profile.

File: PHPgettext/gettext.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ function readintarray($count) {
    * @param object Reader the StreamReader object
    * @param boolean enable_cache Enable or disable caching of strings (default on)
    */
-  function __construct($Reader, $enable_cache = true) {
+  function gettext_reader($Reader, $enable_cache = true) {
```
Comment (seamuslee001) [inline PHPgettext/gettext.php]: @borisson This should be reversed and can you change file permissions back to 664 pls

File: PHPgettext/streams.php [hunk 2]
```diff
@@ -49,7 +49,7 @@ class StringReader {
   var $_pos;
   var $_str;
 
-  function __construct($str='') {
+  function StringReader($str='') {
```
Comment (seamuslee001) [inline PHPgettext/streams.php]: @borisson @mlutfy the changes in this file should be reverted (php7 and more so later versions of php5 prefer the __construct style for class constructors

**Comment by seamuslee001** [comment]
@borisson @mlutfy PR looks ok to me, added some comments around things that should be changed. I don't have a multilingual environment to test this on tho

**Comment by mlutfy** [comment]
@borisson Can you remove the `examples` folder? They include some binary `mo` files (small, but still, may as well avoid it).

**Comment by borisson** [comment]
All requested changes have been done.

**Comment by totten** [comment]
jenkins, test this please

**Comment by borisson** [comment]
Looks like this is failing tests, but I don't understand why. They don't seem to be related.

**Comment by borisson** [comment]
As another option, we could remove the library from this package, and require the https://github.com/phpmyadmin/motranslator/ library trough composer instead. That removes maintenance burden from the CiviCRM maintainers.

It's also a better-tested and more modern version of this package.

I'm not sure what the BC policy about that would be, but I think that this is probably the best solution in the future.

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by seamuslee001** [comment]
This looks fine to me now

**Pull request merged**